prov/verbs/src/windows/verbs_nd_rdma.c

```diff
@@ -136,23 +136,51 @@ int rdma_create_id(struct rdma_event_channel *channel, struct rdma_cm_id **id,
 	id_nd->ov_file = INVALID_HANDLE_VALUE;
 	ofi_atomic_initialize32(&id_nd->cm_events_pending, 0);
 
+	ofi_mutex_init(&id_nd->connect_event.base.lock);
+	pthread_cond_init(&id_nd->connect_event.base.cond, 0);
 	id_nd->connect_event.base.event_cb = nd_cm_connect_ack;
 	id_nd->connect_event.base.error_cb = nd_cm_connect_nack;
 	id_nd->connect_event.id = *id;
 	id_nd->connect_event.channel = channel;
 
+	ofi_mutex_init(&id_nd->peer_event.base.lock);
+	pthread_cond_init(&id_nd->peer_event.base.cond, 0);
 	id_nd->peer_event.base.event_cb = nd_cm_connect_ack;
 	id_nd->peer_event.base.error_cb = nd_cm_connect_nack;
 	id_nd->peer_event.id = *id;
 	id_nd->peer_event.channel = channel;
 
+	ofi_mutex_init(&id_nd->listen_event.base.lock);
+	pthread_cond_init(&id_nd->listen_event.base.cond, 0);
 	id_nd->listen_event.base.event_cb = nd_cm_connreq_event;
 	id_nd->listen_event.base.error_cb = nd_cm_connreq_error;
 	id_nd->listen_event.listen_id = *id;
 
 	return 0;
 }
 
+static void remove_cm_id_events_from_channel(struct rdma_cm_id *id)
+{
+	struct nd_event_channel *ch_nd =
+		container_of(id->channel, struct nd_event_channel, channel);
+	if (!dlistfd_empty(&ch_nd->q)) {
+		struct nd_cm_event *entry_nd;
+		struct dlist_entry *tmp;
```

Comment:
Please declare variables at the top of functions.  It makes them easier to find.  This function would be easier to read as:
```
dlist_foreach_container_safe() {
    ...
}
```
no dlistfd_empty() check.  The foreach should check for an empty queue.